packages/@lwc/engine-core/src/framework/vnodes.ts

```diff
@@ -37,6 +42,17 @@ export interface VStatic extends BaseVNode {
     fragment: Element;
 }
 
+export interface VFragment extends BaseVNode, BaseVParent {
+    // In a fragment elm represents the last node of the fragment,
+    // which is the upper text node delimiter. Used in the updateStaticChildren routine.
```

Comment:
Did you mean `lower` text node delimiter here?
```
<template>
    '' <!-- upper delimiter -->
    ...fragment children
    '' <!-- lower delimiter -->
</template>
```
or am I missing something?

Reply:
😄 , I was thinking about the last text node (the one you think as `lower`), that comment definitely needs work